ant-bootstrap/src/cache_store/cache_data_v1.rs

```diff
@@ -51,17 +53,26 @@ impl CacheData {
                     while addrs.len() > max_addrs_per_peer {
                         addrs.pop_front();
                     }
+                    found_existing = true;
                     break;
                 }
             }
 
-            self.peers.push_back((*other_peer, other_addrs.clone()));
+            if !found_existing {
+                self.peers.push_back((*other_peer, other_addrs.clone()));
+            }
 
-            while self.peers.len() > max_peers {
-                self.peers.pop_front();
+            // break if limit reached
+            if self.peers.len() >= max_peers {
```

Comment:
the previous logic using `self.peers.pop_front();` which indicates:
1. peers are odered from `old to new`
2. once cap reached, `old` ones preferred to be replaced by the later on new ones.

This actually indicates couple of issues in this merging logic:
1.  `self` seems assumed to be always being `old` than the ones from `other` , as using `self.peers.push_back((*other_peer, other_addrs.clone()));`
2.  once cap reached, those potential `new` nodes in other won't be able to be `refreshed` in